views/share/index.js

```diff
@@ -1,11 +1,94 @@
+var async = require('async');
 var view = require('../../lib/view');
 var i18n = require('../../lib/i18n');
 var publish = require('../../lib/publish');
 var page = require('page');
+var config = require('../../config');
+var localforage = require('localforage');
+var xhr = require('xhr');
 var app;
 
 var PUBLISH_TIMEOUT = 20000;
 
+var pre_publish_hooks = [
+    function image_upload(app, user, hook_callback) {
+        var app_data = app.data;
+        var images = app_data.blocks.filter(function (block) {
+            return block.type === 'image' && block.attributes.image.hash;
+        });
+
+        if (images.length === 0) {
+            return hook_callback();
+        }
+
+        var q = async.queue(function (block, q_callback) {
+            var key = 'image/' + block.attributes.image.hash;
+
+            localforage.getItem(key, function (error, file) {
+                if (!file) {
+                    q_callback();
+                    return;
+                }
+
+                // Support dev mode for local publishing
+                var request_body = {};
+                if (config.PUBLISH_DEV_MODE) request_body.user = user;
+
+                xhr({
+                    method: 'POST',
+                    url: config.IMAGE_REQUEST_ENDPOINT,
+                    json: request_body,
+                    withCredentials: true
+                }, function (wp_error, response, data) {
+                    data.starts_with.forEach(function (field) {
```

Comment:
this is dangerous, if data.starts_with doesn't exist. which it doesnt if there's an error